lib/Model.ts

```diff
@@ -725,18 +726,21 @@ export class Model<TDocument extends { _id?: any }, TInstance> {
             })
         }
         else {
-            let instanceType = <InstanceImplementation<MapReducedDocument<Key, Value>, any> & { mapReduceOptions: MapReduceFunctions<TDocument, Key, Value> }>functions;
+            let instanceType = <instance>functions;
             return new Bluebird<void>((resolve, reject) => {
                 if (options.out && options.out == "inline")
                     return reject(new Error("Expected a non-inline mapReduce output mode for this method signature"));
                 let opts = <MongoDB.MapReduceOptions>options;
                 let out : {[op: string]: string} = {};
                 out[(<string>options.out)] = instanceType.collection;
                 opts.out = out;
-                this.collection.mapReduce(instanceType.mapReduceOptions.map, instanceType.mapReduceOptions.reduce, opts, (err, data) => {
-                    if (err) return reject(err);
-                    return resolve();
-                });
+                if (instanceType.mapReduceOptions) {
+                    this.collection.mapReduce(instanceType.mapReduceOptions.map, instanceType.mapReduceOptions.reduce, opts, (err, data) => {
+                        if (err) return reject(err);
+                        return resolve();
+                    });
+                }
+                else return reject("mapReduceOptions not provided");
```

Comment:
Can we reject this with an actual Error object? Strings as errors are generally a bad idea for debugging and handling purposes.